From 393bf23a2683971a72217839657bb2945a36ee54 Mon Sep 17 00:00:00 2001
From: "Azamat H. Hackimov" <azamat.hackimov@gmail.com>
Date: Mon, 3 Jul 2023 14:12:22 +0300
Subject: [PATCH] Fix compilation with Perl 5.38

Perl now includes own `struct object` which clashes with elinks
implementation. Renamed `struct object` to `struct elinks_object` to
avoid it.

Bug: https://bugs.gentoo.org/909042
---
 src/main/object.h  | 6 +++---
 src/protocol/uri.c | 2 +-
 src/protocol/uri.h | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

Index: elinks-0.16.1.1/src/main/object.h
===================================================================
--- elinks-0.16.1.1.orig/src/main/object.h
+++ elinks-0.16.1.1/src/main/object.h
@@ -11,7 +11,7 @@ extern "C" {
 #define DEBUG_REFCOUNT
 #endif
 
-struct object {
+struct elinks_object {
 	int refcount;
 #ifdef CONFIG_DEBUG
 	char *name;
@@ -20,10 +20,10 @@ struct object {
 
 #define OBJECT_HEAD(type)						\
 	LIST_HEAD(type);						\
-	struct object object
+	struct elinks_object object
 
 struct object_head {
-	OBJECT_HEAD(struct object *);
+	OBJECT_HEAD(struct elinks_object *);
 };
 
 #ifdef DEBUG_REFCOUNT
Index: elinks-0.16.1.1/src/protocol/uri.c
===================================================================
--- elinks-0.16.1.1.orig/src/protocol/uri.c
+++ elinks-0.16.1.1/src/protocol/uri.c
@@ -1578,7 +1578,7 @@ struct uri_cache_entry {
 
 struct uri_cache {
 	struct hash *map;
-	struct object object;
+	struct elinks_object object;
 };
 
 static struct uri_cache uri_cache;
Index: elinks-0.16.1.1/src/protocol/uri.h
===================================================================
--- elinks-0.16.1.1.orig/src/protocol/uri.h
+++ elinks-0.16.1.1/src/protocol/uri.h
@@ -89,7 +89,7 @@ struct uri {
 	unsigned int form:1;	/* URI originated from form */
 
 	/* Usage count object. */
-	struct object object;
+	struct elinks_object object;
 };
 
 enum uri_errno {
